Source/WebInspectorUI/UserInterface/Views/ComputedStyleDetailsPanel.js

```diff
@@ -225,59 +221,85 @@ WI.ComputedStyleDetailsPanel = class ComputedStyleDetailsPanel extends WI.StyleD
             if (styleSection === this._computedStyleSection)
                 continue;
 
-            // WI.ComputedStyleSection is a view but its element is attached to a WI.DetailsSection which isn't. Reparent it before removing.
-            // FIXME: <https://webkit.org/b/152269> - Web Inspector: Convert DetailsSection classes to use View
-            this.element.appendChild(styleSection.element);
-            this.removeSubview(styleSection);
-            styleSection.element.remove();
-            this._detailsSectionByStyleSectionMap.delete(styleSection);
-            detailsSection.element.remove();
+            this._removeVariablesStyleSection(styleSection);
         }
 
-        this._variablesStyleSectionForGroupTypeMap.clear();
+        if (!this.nodeStyles.computedStyle)
+            return;
 
         switch (this.variablesGroupingMode) {
         case WI.ComputedStyleDetailsPanel.VariablesGroupingMode.Ungrouped:
-            this._renderVariablesStyleSectionGroup(this.nodeStyles.computedStyle, WI.DOMNodeStyles.VariablesGroupType.Ungrouped);
+            this._createVariablesStyleSection(WI.CSSStyleDeclaration.VariablesGroupType.Ungrouped);
             break;
 
         case WI.ComputedStyleDetailsPanel.VariablesGroupingMode.ByType:
-            for (let [type, style] of this.nodeStyles.variableStylesByType)
-                this._renderVariablesStyleSectionGroup(style, type, WI.ComputedStyleDetailsPanel.displayNameForVariablesGroupType(type));
+            for (let type of Object.values(WI.CSSStyleDeclaration.VariablesGroupType)) {
+                if (type === WI.CSSStyleDeclaration.VariablesGroupType.Ungrouped)
+                    continue;
+
+                this._createVariablesStyleSection(type, WI.ComputedStyleDetailsPanel.displayNameForVariablesGroupType(type));
+            }
             break;
         }
 
         if (this._filterText)
             this.applyFilter(this._filterText);
     }
 
+    didLayoutSubtree()
+    {
+        super.didLayoutSubtree();
+
+        for (let [styleSection, detailsSection] of this._detailsSectionByStyleSectionMap) {
+            detailsSection?.element.classList.toggle("hidden", styleSection.isEmpty);
```

Comment:
is it actually possible for `detailsSection` to be `null`/`undefined`?  i'd think that everything in the `Map` would always exist

Reply:
Yes, that's true. This leftover from copy/pasta